src/reports/models/reportsManager.js

```diff
@@ -88,11 +92,30 @@ module.exports.postStats = async (report, stats) => {
     }
     await databaseConnector.updateReport(report.test_id, report.report_id, { phase: report.phase, last_updated_at: statsTime });
     report = await module.exports.getReport(report.test_id, report.report_id);
+    await updateReportBenchMarkIfNeeded(report);
     notifier.notifyIfNeeded(report, stats);
 
     return stats;
 };
 
+async function updateReportBenchMarkIfNeeded(report) {
+    if (!reportUtil.isAllRunnersInExpectedPhase(report, constants.SUBSCRIBER_DONE_STAGE)) {
+        return;
+    }
+    const testBenchMarkData = await extractBenchMarkData(report.test_id);
+    if (testBenchMarkData) {
+        const reportAggregate = await aggregateReportManager.aggregateReport(report);
+        const reportBenchMark = benchmarkCalculator.calculate(testBenchMarkData, reportAggregate.aggregate);
+        const { data, score } = reportBenchMark;
+        await databaseConnector.updateReportBenchMark(report.test_id, report.report_id, score, JSON.stringify(data));
+    }
+}
+
+async function extractBenchMarkData(testId) {
+    const res = await testDbConnector.getTestBenchMark(testId);
```

Comment:
reports shouldn't go to tests DB. it should go to tests manage.
test manager should return the connector as JSON

please rename res to benchmark

Reply:
fixed